### PR TITLE
Don't allow groups without permissions.

### PR DESCRIPTION
**Pull request by erikgrinaker**
It probably shouldn't be possible to create a group without specifying a permissions list. If it should, then it's probably better to default to an empty slice instead of a `nil` pointer. Related to #335.

This PR rejects groups that have a `nil` permissions list. This causes a `500` to the client, since it just returns a plain `errors.Error` struct - it should probably return some sort of `proto.ValidationError` which is mapped to `400` in `proto/errors.go`, but it's a bit unclear how I can instantiate a standard error which is interpreted as a `proto.ValidationError`.

### CODE REVIEW

**Comment by atombender** [comment]
👍  You can create a new error type and then check for it in `proto/protofy/errors.go`. E.g., just declare a string error:

```go
type InvalidGrant string
func (err InvalidGrant) Error() string {
  return string(err)
}
```

**Comment by erikgrinaker** [comment]
Since the errors are wrapped with `errors.Wrap()` further up the stack, I had to add a `errors.Cause()` call to unwrap the original error in `protofy.DoError`. I believe this is the correct behavior in any case, but errors which earlier were wrapped (and thus handled as generic, untyped errors) may now be handled as their actual type, changing their returned HTTP status code.

File: pkg/security/models.go [hunk 1]
```diff
@@ -24,14 +24,17 @@ func (g Grant) Validate() error {
 		// Try compiling
 		expr, err := gradientql.Parse(fmt.Sprintf("*[%s]", g.Filter), gradientql.Params{"identity": "dummy"})
 		if err != nil {
-			return errors.Wrapf(err, "Error while parsing group filter %v", g.Filter)
```
Comment (atombender) [inline pkg/security/models.go]: You should use `%s` for the error, not `%v` (also below), otherwise it might dump some internal Go structure.

Comment (erikgrinaker) [reply pkg/security/models.go]: Well, `error.Error()` always returns a string by definition, but sure - fixed.

Comment (atombender) [reply pkg/security/models.go]: True, I kind of misread it and thought you were passing just `err` (which works fine with `%s`). But `%v` also quotes the string.

Comment (erikgrinaker) [reply pkg/security/models.go]: Nah - `%q` quotes the string, `%v` is just an "appropriate representation" of whatever type is passed.

Comment (atombender) [reply pkg/security/models.go]: Hm, I'm probably thinking of `%#v`.

**Pull request merged**